tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -15,15 +15,325 @@
 
 #include "s2n_test.h"
 
+#include "testlib/s2n_testlib.h"
+
+#include "tls/extensions/s2n_key_share.h"
+#include "tls/extensions/s2n_server_supported_versions.h"
+
+#include "tls/s2n_cipher_preferences.h"
+#include "tls/s2n_cipher_suites.h"
 #include "tls/s2n_tls.h"
+#include "tls/s2n_tls13.h"
+#include "tls/s2n_tls13_handshake.h"
+
+#include "tls/extensions/s2n_server_key_share.h"
+
 #include "error/s2n_errno.h"
 
+const uint8_t SESSION_ID_SIZE = 1;
+const uint8_t COMPRESSION_METHOD_SIZE = 1;
+
+/* from RFC: https://tools.ietf.org/html/rfc8446#section-4.1.3*/
+const uint8_t hello_retry_request_random_test_buffer[S2N_TLS_RANDOM_DATA_LEN] = {
+    0xCF, 0x21, 0xAD, 0x74, 0xE5, 0x9A, 0x61, 0x11, 0xBE, 0x1D, 0x8C, 0x02, 0x1E, 0x65, 0xB8, 0x91,
+    0xC2, 0xA2, 0x11, 0x16, 0x7A, 0xBB, 0x8C, 0x5E, 0x07, 0x9E, 0x09, 0xE2, 0xC8, 0xA8, 0x33, 0x9C
+};
```

Comment:
The reason I repeat constants like this one is to catch fat fingers, bad find/replaces, or mistakes during refactors later. This way we can rely on the test instead of a code review picking up an accidental change.